src/sequelize.js

```diff
@@ -282,6 +282,11 @@ export class Sequelize extends SequelizeTypeScript {
       this.options.dialect = 'postgres';
     }
 
+    //     if (this.options.define.hooks) {
```

Comment:
Commented out until I figure out what to do with this. Opinion?

Reply:
Not sure if I understand this question. Can you give some more context?

Reply:
For sure!

Users can use the `define` Sequelize option to pre-configure models. That includes `hooks`:

```ts
new Sequelize({
  define: {
    hooks: {
      // adds "beforeFind" hook to all models
      beforeFind: console.log,
    },
  },
});
```

The problem(?) is that currently, that hook will not be added if the model adds its own hook of the same type:

```ts
// This model will have both a beforeSave & beforeFind hook
sequelize.define({}, {
  hooks: {
    beforeSave: console.log,
  },
})

// This model will only have this beforeFind hook, not the one declared in the Sequelize "define" options
sequelize.define({}, {
  hooks: {
    beforeFind: console.log,
  },
})
```

So our options are:

1. Leave it like that: Sounds super counter-intuitive that your hook doesn't get added if you add another hook when defining your model
2. Remove the ability to add hooks through the `define` option: You can already listen to all model events by adding that listener on your Sequelize instance:
   ```ts
   new Sequelize({
     hooks: {
       // listens to this hook on all models
       beforeFind: console.log,
     },
   });
   ```
   The only difference is that since this hook is listened to on the Sequelize instance, instead of being added to every model, it cannot be removed from individual models. But I would also argue that adding the listener to all models then removing it from some of them means your code needs to be reworked
3. Preserve the ability to add hooks through the `define` option, but remove the part where your hook won't be added if your model define its own hook.
   ```ts
   // This adds *both* the hook declared through the "define" option, and the one declared here
   sequelize.define({}, {
     hooks: {
       beforeFind: console.log,
     },
   })
   ```

Reply:
Option 3 adds both hooks, which I think is what users would typically expect, since you already can add multiple hooks of the same type on the same model.

If it's counter-intuitive to the point where we need to warn users about it, I think it would then be better to remove it

Reply:
Done. The options specified in the `define` global option are merged with the ones specified in `sequelize.define`, `model.init` and decorators following the same logic that decorators use to merge their options